### PR TITLE
Make scriptblock-based calculated properties work again in ConvertTo-Html

### LINKED ISSUE #6993: Passing calculated properties to ConvertTo-Html is broken
Steps to reproduce
------------------

```powershell
'one', 'two' | ConvertTo-Html -Property @{ l='Name'; e={ $_ } }
```

Expected behavior
-----------------

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>HTML TABLE</title>
</head><body>
<table>
<colgroup><col/></colgroup>
<tr><th>Name</th></tr>
<tr><td>one</td></tr>
<tr><td>two</td></tr>
</table>
</body></html>
```

This is how it has always worked in Windows PowerShell.

The property name becomes the column header (`th`), and the values the column data (`td`).

Actual behavior
---------------

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>HTML TABLE</title>
</head><body>
<table>
<colgroup><col/></colgroup>
<tr><th>Name</th></tr>
<tr><td></td></tr>
<tr><td></td></tr>
</table>
</body></html>
```

Note how the `<td>` elements are unexpectedly empty.

Environment data
----------------

```powershell
PowerShell Core v6.1.0-preview.2 on macOS 10.13.4
PowerShell Core v6.1.0-preview.2 on Ubuntu 16.04.4 LTS
PowerShell Core v6.1.0-preview.2 on Microsoft Windows 10 Pro (64-bit; Version 1709, OS Build: 16299.371)
```

### PR DESCRIPTION
**Pull request by mklement0**
## PR Summary

Fixes #6993

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/convert-HTML.cs [hunk 1]
```diff
@@ -341,10 +341,20 @@ private void InitializeResolvedNameMshParameters()
...
                     Hashtable ht = CreateAuxPropertyHT(label, alignment, width);
-                    ht.Add(FormatParameterDefinitionKeys.ExpressionEntryKey, resolvedName.ToString());
+                    if (resolvedName.Script != null)
+                    {
+                        // script block
+                        ht.Add(FormatParameterDefinitionKeys.ExpressionEntryKey, resolvedName.Script);
+                    }
+                    else
+                    {
+                        // property name
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/convert-HTML.cs]: I'd remove the obvious comment and enhance comment in line 346.

**Review by iSazonov — COMMENTED** [review]
@mklement0 Please add a commit with `[Feature]` in title to run full test set.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/convert-HTML.cs (hunk 1 above)
Comment (mklement0) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/convert-HTML.cs]: @iSazonov: Done. Is it sufficient to add "[Feature]" in the first line of a commit message _once_?

**Review by iSazonov — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
@mklement0 Sorry, I lost the PR :-( Do we ready to merge?

**Comment by iSazonov** [comment]
@mklement0  Maintainer can squash and merge as needed. Thanks for your contribution!

**Comment by mklement0** [comment]
That's good to know, @iSazonov - thanks for merging.

**Pull request merged**